tests/fuzz/security_fuzzer.go

```diff
@@ -0,0 +1,32 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fuzz
+
+import (
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+
+	"istio.io/istio/security/pkg/pki/util"
+)
+
+func FuzzGenCSR(data []byte) int {
+	f := fuzz.NewConsumer(data)
+	certOptions := util.CertOptions{}
+	err := f.GenerateStruct(&certOptions)
+	if err != nil {
+		return 0
+	}
+	_, _, _ = util.GenCSR(certOptions)
```

Comment:
One concern: I think this is fairly expensive call (in CPU time) so we may end up not getting many iterations

Reply:
Locally it runs alright. I will keep an eye out for the stats from runs by OSS-fuzz.